ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/IcebergTableMaintainer.java

```diff
@@ -183,14 +183,14 @@ private void expireSnapshots(long olderThan, Set<String> exclude) {
         .expireOlderThan(olderThan)
         .deleteWith(
             file -> {
-              try {
+              try (ArcticFileIO io = arcticFileIO()) {
```

Comment:
I don't think introducing try-with-resources here is a good approach. The IO objects in the table are not suitable to be closed every time they are used, even though closing them now does nothing.

I understand that IDEA would currently raise a warning here. Perhaps we can try to find some other methods to avoid reporting an error.